pkg/webhook/server.go

```diff
@@ -103,3 +111,40 @@ func (s *Server) setCert(name, namespace string) error {
 
 	return s.onCertChange(webhookCert)
 }
+
+func (s *Server) certCommonName() string {
+	return fmt.Sprintf("%s.%s.svc", s.name, s.namespace)
+}
+
+// configureCertificateRotation gets the certificate from the certificate manager and spawns a goroutine to watch for certificate rotation.
+func (s *Server) configureCertificateRotation() func() {
+	var once sync.Once
+	stop := make(chan struct{})
+	certRotationChan, unsubscribeRotation := s.cm.SubscribeRotations(s.certCommonName())
+
+	// Handle the rotations
+	go func() {
+		log.Info().Str("webhook", s.name).Str("cn", s.certCommonName()).Msg("Listening for certificate rotations")
+		for {
+			select {
+			case <-certRotationChan:
+				log.Debug().Str("webhook", s.name).Str("cn", s.certCommonName()).Msg("Certificate rotation was initiated for webhook")
+				if err := s.setCert(); err != nil {
+					// TODO(#4956): raise event for failures
+					log.Error().Err(err).Str("webhook", s.name).Msgf("Error rotating certificate for webhook")
+				}
+				log.Info().Str("webhook", s.name).Str("cn", s.certCommonName()).Msg("Certificate rotated for webhook")
+			case <-stop:
```

Comment:
We'll need to pass in a new stop as well, since we're no longer returning a cancel func. Using the context from Run could work

Reply:
I don't think we can use the context from the Run (this is what I initially did) because it will miss the cert rotations since the cert is set during the call to `NewWebhook`. One option is to move the initial cert creation to run but this changes the current behavior (https://github.com/openservicemesh/osm/pull/4952/files#r935739443).  

Should we change that behavior and panic if we can call [Tresor](https://github.com/openservicemesh/osm/pull/4952/files#r937169543)? In the case of certmanager and other external system this could happen pretty regularly and retries might be a better option with a healthprobe.

Reply:
could remove run, and call go run() from within NewServer. 

Just a thought